### PR TITLE
CAPZ: Release v29.1.0.

### PR DESCRIPTION
**Pull request by mcharriere**
Towards https://github.com/giantswarm/giantswarm/issues/31626

Signed-off-by: Matias Charriere <matias@giantswarm.io>

### Checklist

- [x] Roadmap issue created
- [x] Release uses latest stable Flatcar
- [x] Release uses latest Kubernetes patch version

### Triggering E2E tests

To trigger the E2E test for each new Release added in this PR, add a comment with the following:

`/run releases-test-suites`

If you want to trigger conformance tests, you can do so by adding a comment similar to the following:

`/run conformance-tests PROVIDER=capa RELEASE_VERSION=29.1.0`

For more details see the [README.md](/README.md#running-tests-against-prs).

### CODE REVIEW

File: azure/v29.0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# :zap: Giant Swarm Release v29.0.1 for Azure :zap:
+
+## Changes compared to v29.0.0
+
+### Apps
+
+- coredns from v1.21.0 to v1.22.0
+- cert-exporter from v2.9.1 to v2.9.2
+- observability-bundle from v1.5.1 to v1.6.1
```
Comment (mcharriere) [inline azure/v29.0.1/README.md:9]: I remember that it was mentioned that we don't want to bump a minor version in a patch release, but is that so?

Comment (mcharriere) [reply azure/v29.0.1/README.md:9]: from version is wrong
```suggestion
- observability-bundle from v1.5.3 to v1.6.1
```

**Comment by Gacko** [comment]
I know, this is still a WIP, but can you elaborate on why there are minor app bumps while the release itself is just a patch release? I think we went for a minor release in CAPA to bump `observability-bundle` and others in a minor step.

**Comment by mcharriere** [comment]
> From the name of the PR branch I assume you're forging this mostly for updating CoreDNS. Since this is a minor bump of a v1.x.x, I'd rather head for a CAPZ v29.1.0.

Yeah, I'm fine with that. Let me rework this PR then.

**Comment by mcharriere** [comment]
This PR updates:
- coredns from v1.21.0 to v1.22.0. This is what I need.
- cert-exporter from v2.9.1 to v2.9.2. It's in `requests`.
- observability-bundle from v1.5.3 to v1.6.1. It's in `requests`.

Not sure if we want to mark v29.0.0 as deprecated in this same PR.

cc @alex-dabija

File: azure/v29.1.0/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Changes compared to v29.0.0
+
+### Apps
+
+- coredns from v1.21.0 to v1.22.0
+- cert-exporter from v2.9.1 to v2.9.2
+- observability-bundle from v1.5.3 to v1.6.1
+
+### coredns [1.22.0](https://github.com/giantswarm/coredns-app/releases/tag/v1.22.0)
```
Comment (Gacko) [inline azure/v29.1.0/README.md:11]:
```suggestion
### coredns [v1.21.0...v1.22.0](https://github.com/giantswarm/coredns-app/compare/v1.21.0...v1.22.0)
```

File: azure/v29.1.0/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+#### Changed
+
+- Update `coredns` image to [1.11.3](https://github.com/coredns/coredns/releases/tag/v1.11.3).
+
+#### Removed
+
+- Removed legacy Giant Swarm monitoring labels as coredns is monitored through a prometheus-operator generated servicemonitor.
+
+### cert-exporter [2.9.2](https://github.com/giantswarm/cert-exporter/releases/tag/v2.9.2)
```
Comment (Gacko) [inline azure/v29.1.0/README.md:21]:
```suggestion
### cert-exporter [v2.9.1...v2.9.2](https://github.com/giantswarm/cert-exporter/compare/v2.9.1...v2.9.2)
```

File: azure/v29.1.0/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+#### Added
+
+- Chart: Add VPA and resources configuration for deployment and daemonset. ([#382](https://github.com/giantswarm/cert-exporter/pull/382))
+
+### observability-bundle [1.6.1](https://github.com/giantswarm/observability-bundle/releases/tag/v1.6.1)
```
Comment (Gacko) [inline azure/v29.1.0/README.md:27]:
```suggestion
### observability-bundle [v1.5.3...v1.6.1](https://github.com/giantswarm/observability-bundle/compare/v1.5.3...v1.6.1)
```

File: azure/v29.1.0/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+#### Fixed
+- Disable usage reporting to GrafanaLabs by:
+  - Bumping `alloyLogs` and `alloyMetrics` to v0.4.1.
+  - Bumping `grafanaAgent` to v0.4.6.
```
Comment (Gacko) [inline azure/v29.1.0/README.md:32]:
```suggestion
#### Added

- Add `alloy` v0.4.0 as `alloyMetrics`.

#### Changed

- Disable usage reporting to GrafanaLabs by:
  - Bumping `alloyLogs` and `alloyMetrics` to v0.4.1.
  - Bumping `grafanaAgent` to v0.4.6.
```

**Review by Gacko — COMMENTED** [review]
Left some suggestions. I'll check if there are any other components to update from our side tomorrow.

**Comment by alex-dabija** [comment]
> Not sure if we want to mark v29.0.0 as deprecated in this same PR.

@mcharriere Yes, please also deprecate the v29.0.0 release because we want to maintain as few releases as possible.

File: azure/v29.1.0/release.yaml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    version: 0.1.1
+    catalog: cluster
+    dependsOn:
+    - cilium
+  - name: node-exporter
+    version: 1.19.0
+    dependsOn:
+    - kyverno-crds
+  - name: observability-bundle
+    version: 1.6.1
```
Comment (TheoBrigitte) [inline azure/v29.1.0/release.yaml:83]: Can we have observability-bundle v1.6.2 here ?
```suggestion
  - name: observability-bundle
    version: 1.6.2
```

**Review by Gacko — CHANGES_REQUESTED** [review]
Please consider [this Slack thread](https://gigantic.slack.com/archives/C02HLSDH3DZ/p1726691243035579).

Therefore I'd like to hold this PR for now so we can focus on patching older releases and get this PR here aligned to the existing CAPA v29.1.0 release afterwards. Probably next week.

**Review by Gacko — APPROVED** [review]

**Comment by Gacko** [comment]
I introduced the mentioned changes to the others apps, especially `cluster-aws`, which ships some important fixes, Kubernetes and Flatcar.

See the `README.md` and `release.diff` for what changes compared to v29.0.0.

**Comment by Gacko** [comment]
Regarding v29.0.0 deprecation: Let's please do this in a separate PR.

**Review by stone-z — APPROVED** [review]

**Comment by Gacko** [comment]
Tests are currently failing because of Teleport Kube Agent. @ssyno is already working on fixing it. We therefore crafted [this draft PR](https://github.com/giantswarm/releases/pull/1420) for testing changes. Once the app got fixed and released, we will integrate the new version in this PR here.

**Review by ssyno — APPROVED** [review]

**Pull request merged**